crates/biome_js_analyze/src/lint/nursery/use_unique_element_ids.rs

```diff
@@ -63,18 +63,41 @@ declare_node_union! {
 }
 
 impl UseUniqueElementIdsQuery {
-    fn find_id_attribute(&self, model: &SemanticModel) -> Option<IdProp> {
+    fn create_element_call(&self, model: &SemanticModel) -> Option<ReactCreateElementCall> {
         match self {
-            Self::AnyJsxElement(jsx) => jsx.find_attribute_by_name("id").map(IdProp::from),
             Self::JsCallExpression(expression) => {
-                let react_create_element =
-                    ReactCreateElementCall::from_call_expression(expression, model)?;
-                react_create_element
-                    .find_prop_by_name("id")
-                    .map(IdProp::from)
+                ReactCreateElementCall::from_call_expression(expression, model)
+            }
+            &Self::AnyJsxElement(_) => None,
+        }
+    }
+
+    fn element_name(&self, model: &SemanticModel) -> Option<Box<str>> {
```

Comment:
You want `TokenText` to avoid the string allocation.

Reply:
Thanks! I knew there should be something, but didn't realize it's so easy